src/containers/Application/ApplicationSubmission.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect } from 'react'
+import { Button, Header, Icon, Label, List, Message, Segment } from 'semantic-ui-react'
+import { Loading } from '../../components'
+import strings from '../../utils/constants'
+import { useRouter } from '../../utils/hooks/useRouter'
+import { useUserState } from '../../contexts/UserState'
+import useGetApplicationStatus from '../../utils/hooks/useGetApplicationStatus'
+import useLoadApplication from '../../utils/hooks/useLoadApplication'
+import { ApplicationStatus } from '../../utils/generated/graphql'
+import { Link } from 'react-router-dom'
+
+const ApplicationSubmission: React.FC = () => {
+  const {
+    userState: { currentUser },
+  } = useUserState()
+
+  const { query, push } = useRouter()
+  const { serialNumber } = query
+
+  const { error, loading, application, isApplicationLoaded } = useLoadApplication({
+    serialNumber: serialNumber as string,
+  })
+
+  const {
+    error: statusError,
+    loading: statusLoading,
+    appStatus,
+    appStages,
+  } = useGetApplicationStatus({
+    serialNumber: serialNumber as string,
+    isApplicationLoaded,
+  })
+
+  useEffect(() => {
+    if (statusLoading) return
+    // Check if application is in Draft or Changes required status and redirect to the summary page
+    // Note: The summary page has its own redirection logic to ay specific page (with invalid items).
+    if (
+      appStatus?.status === ApplicationStatus.Draft ||
+      appStatus?.status === ApplicationStatus.ChangesRequired
+    ) {
+      push(`/application/${serialNumber}/summary`)
+    }
+  }, [appStatus])
```

Comment:
The problem is somewhere in here. The "appStatus" is still "Draft", so it's re-directing back to Summary.

It shouldn't be allowed to acquire the status until all Triggers are reset, so it's jumping the gun here somehow. Will investigate further...

Reply:
Okay -- it's a cache issue. Even when the Status has been set to "COMPLETED" in database, this hook is still returning the cached value of "DRAFT". Need to force a re-fetch from server

Reply:
I guess this is fixed with by bug fix done in [5fad684](https://github.com/openmsupply/application-manager-web-app/pull/234/commits/5fad684b765ea8317e31410ebaa1a5a78b84b78c)?

Reply:
Yup! 👍🏼